src/clr-angular/data/datagrid/datagrid-column-reorder-droppable.spec.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component } from '@angular/core';
+import { ClrDatagridColumnReorderDroppable } from './datagrid-column-reorder-droppable';
+import { TestContext } from '../../utils/testing/helpers.spec';
+import { ColumnOrderModelService } from './providers/column-order-model.service';
+import { TableSizeService } from './providers/table-size.service';
+import { DomAdapter } from '../../utils/dom-adapter/dom-adapter';
+import { MOCK_TABLE_SIZE_PROVIDER, MockTableSizeService } from './providers/table-size.service.mock';
+
+import {
+  MOCK_COLUMN_ORDER_MODEL_PROVIDER,
+  MockColumnOrderModelService,
+  populateMockProps,
+} from './providers/column-order-model.service.mock';
+import { ColumnHeaderSides } from './enums/header-sides.enum';
+
+@Component({
+  template: `<clr-dg-column-reorder-droppable [side]="side"></clr-dg-column-reorder-droppable>`,
+})
+class TestComponent {
+  side: ColumnHeaderSides;
+}
+
+export default function(): void {
+  describe('ClrDatagridColumnReorderDroppable component', function() {
+    let context: TestContext<ClrDatagridColumnReorderDroppable, TestComponent>;
+    let componentInstance: TestComponent;
+    let reorderDroppable: ClrDatagridColumnReorderDroppable;
+    let columnOrderModelService: MockColumnOrderModelService;
+    let tableSizeService: MockTableSizeService;
+
+    beforeEach(function() {
+      const PROVIDERS_NEEDED = [MOCK_TABLE_SIZE_PROVIDER, MOCK_COLUMN_ORDER_MODEL_PROVIDER, DomAdapter];
+
+      context = this.create(ClrDatagridColumnReorderDroppable, TestComponent, PROVIDERS_NEEDED);
+      componentInstance = context.fixture.componentInstance;
+      reorderDroppable = context.clarityDirective;
+      columnOrderModelService = context.getClarityProvider(ColumnOrderModelService);
+      tableSizeService = context.getClarityProvider(TableSizeService);
+
+      columnOrderModelService.nextVisibleColumnModel = new MockColumnOrderModelService();
+      columnOrderModelService.previousVisibleColumnModel = new MockColumnOrderModelService();
+
+      populateMockProps(columnOrderModelService.previousVisibleColumnModel, 'dg-mock-group-id', 0, 400);
+      populateMockProps(columnOrderModelService, 'dg-mock-group-id', 1, 200);
+      populateMockProps(columnOrderModelService.nextVisibleColumnModel, 'dg-mock-group-id', 2, 300);
+
+      context.detectChanges();
+    });
+
+    afterEach(function() {
+      context.fixture.destroy();
```

Comment:
You are calling `populateMockProps` in your `beforeEach()`, which create DOM elements that Angular doesn't know about. You need to remove them yourself in this `afterEach()`.

Reply:
Now removing the elements in `afterEach()`.